### PR TITLE
Payment page

### PR DESCRIPTION
**Pull request by sandrine-ds**
https://www.figma.com/file/feooOYGqB6WNGh7ztU0gpT/Checkout-payment-page?type=design&node-id=1039-2788&mode=design&t=UtlA3TJnEreHWWjO-0

### CODE REVIEW

File: clients/payment/src/App.tsx [hunk 1]
```diff
@@ -1,35 +1,79 @@
...
-        <ClientProvider value={unauthenticatedClient}>
-          {match(route)
-            .with({ name: "PaymentLink" }, () => <PaymentPage />)
-            .with({ name: "Success" }, () => <SuccessPage />)
-            .with(P.nullish, () => <NotFoundPage />)
-            .exhaustive()}
-        </ClientProvider>
-      </Suspense>
-    </ErrorBoundary>
+const UnauthenticatedClient = () => {
```
Comment (bloodyowl) [inline clients/payment/src/App.tsx:27]: we could move this component to another file and call it `PaymentArea`

File: clients/payment/src/App.tsx [hunk 2]
```diff
@@ -1,35 +1,79 @@
...
+  const route = Router.useRoute(["Form", "Success"]);
```
Comment (bloodyowl) [inline clients/payment/src/App.tsx:28]: we could have an `*Area` route with `paymentLinkId` and only run the following query when matched

File: clients/payment/src/App.tsx [hunk 3]
```diff
@@ -1,35 +1,79 @@
...
+  const paymentLinkId = route?.params.paymentLinkId;
+
+  const { data } = useUrqlQuery(
+    {
+      query: GetMerchantPaymentLinkDocument,
+      pause: isNullishOrEmpty(paymentLinkId),
```
Comment (bloodyowl) [inline clients/payment/src/App.tsx:34]: only running the query when matching the area route would enable us not adding this condition

File: clients/payment/src/pages/PaymentPage.tsx [hunk 4]
```diff
@@ -1,50 +1,481 @@
...
+  city: string;
+  postalCode: string;
+  state: string;
+};
+
+type Props = {
+  paymentLink: NonNullable<GetMerchantPaymentLinkQuery["merchantPaymentLink"]>;
+};
+
+const sepaNonEeaCountries = ["CHE", "VAT", "GBR", "MCO", "SMR", "AND"];
```
Comment (bloodyowl) [inline clients/payment/src/pages/PaymentPage.tsx:88]: nit: could be nice to have the API provide those for us, hardcoding isn't ideal

File: clients/payment/src/pages/PaymentPage.tsx [hunk 5]
```diff
@@ -1,50 +1,481 @@
...
+            paymentLinkId: paymentLink.id,
           },
-        )
-        .otherwise(() => (
-          <ErrorView />
-        ))}
-    </>
+        })
+          .mapOk(data => data.addSepaDirectDebitPaymentMandateFromPaymentLink)
+          .mapOkToResult(data => (isNotNullish(data) ? Result.Ok(data) : Result.Error(undefined)))
```
Comment (bloodyowl) [inline clients/payment/src/pages/PaymentPage.tsx:195]: `addSepaDirectDebitPaymentMandateFromPaymentLink` is nullable? 🤔

File: clients/payment/src/pages/PaymentPage.tsx [hunk 6]
```diff
@@ -1,50 +1,481 @@
...
+          .mapOkToResult(filterRejectionsToResult)
+          .tapOk(data => {
+            setMandateUrl(data.paymentMandate.mandateDocumentUrl);
+            Router.replace("Success", { paymentLinkId: paymentLink.id });
+          })
+          .tapError(error => {
+            // Router.replace("Error");
+            console.log(error);
```
Comment (bloodyowl) [inline clients/payment/src/pages/PaymentPage.tsx:203]: small `console.log` kept here, also about the commented bit, are we supposed to route to error or leave the form as it is?

File: clients/payment/src/pages/PaymentPage.tsx (hunk 5 above)
Comment (sandrine-ds) [reply clients/payment/src/pages/PaymentPage.tsx:199]: Yes, at the moment it is, but it shouldn't be. Quentin is looking at this

File: clients/payment/src/pages/PaymentPage.tsx (hunk 4 above)
Comment (sandrine-ds) [reply clients/payment/src/pages/PaymentPage.tsx:88]: I spoke to the backend team about this

File: clients/payment/src/pages/PaymentPage.tsx [hunk 7]
```diff
@@ -0,0 +1,503 @@
...
+      <Space height={32} />
+
+      <Field name="paymentMethod">
+        {({ value, onChange }) => (
+          <LakeLabel
+            style={desktop ? styles.segmentedControlDesktop : styles.segmentedControl}
+            label={t("paymentLink.paymentMethod")}
+            render={() => (
+              <SegmentedControl
+                mode="desktop"
```
Comment (ncomont) [inline clients/payment/src/pages/PaymentPage.tsx:306]: shouldn't it be mobile when !desktop ?

**Comment by ncomont** [comment]
The Cancel button is truncated on iPhone

image

**Comment by ncomont** [comment]
The dropdown focus state style is missing

image

**Comment by ncomont** [comment]
We can't dismiss this dropdown with esc key

image

**Comment by ncomont** [comment]
The logo at top should be the vendor logo
image

**Comment by ncomont** [comment]
With these data, the form seems valid, but I get a ValidationRejection 
image
image

**Comment by ncomont** [comment]
💡  Do you think possible to use the validation message instead of a generic one ?
image

**Comment by ncomont** [comment]
image
when an error occurs, no toast is shown

File: clients/payment/src/assets/imgs/logo-swan.svg
Comment (zoontek) [inline clients/payment/src/assets/imgs/logo-swan.svg:1]: Doesn't seem to be used

File: clients/payment/src/pages/ErrorPage.tsx [hunk 8]
```diff
@@ -28,15 +32,25 @@ const styles = StyleSheet.create({
...
+export const ErrorPage = ({ paymentLink }: Props) => {
   return (
     <Box
       direction="column"
       alignItems="center"
       justifyContent="spaceBetween"
       style={styles.container}
     >
-      <SwanLogo />
+      {isNotNullish(paymentLink.merchantProfile.merchantLogoUrl) ? (
```
Comment (zoontek) [inline clients/payment/src/pages/ErrorPage.tsx:47]: ultra nit: you can create small variables for readability

```ts
const { merchantProfile } = paymentLink;
const { merchantName } = merchantProfile;

return (
  // …
```

File: clients/payment/src/pages/ExpiredPage.tsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+export const ExpiredPage = ({ paymentLink }: Props) => {
+  return (
+    <Box
+      direction="column"
+      alignItems="center"
+      justifyContent="spaceBetween"
+      style={styles.container}
+    >
+      {isNotNullish(paymentLink.merchantProfile.merchantLogoUrl) ? (
```
Comment (zoontek) [inline clients/payment/src/pages/ExpiredPage.tsx:45]: Same here

**Pull request closed without merging**